tests/unit/sproxyd.js

```diff
@@ -282,21 +341,86 @@ describe('Sproxyd client', () => {
 
     describe('Get request uid', () => {
         const uids = 'id1:id2:id3';
-        const log = clientNonImmutable.createLogger(uids);
+        const log = client.createLogger(uids);
         const ids = log.getUids();
 
         it('should return first request id without colon', () => {
-            const firstRequestUid = clientNonImmutable._getFirstReqUid(log);
+            const firstRequestUid = client._getFirstReqUid(log);
             assert.notStrictEqual(firstRequestUid, undefined);
             assert.strictEqual(firstRequestUid.indexOf(':'), -1);
             assert.strictEqual(ids[0], firstRequestUid);
         });
 
         it('should return last request id without colon', () => {
-            const lastRequestUid = clientNonImmutable._getLastReqUid(log);
+            const lastRequestUid = client._getLastReqUid(log);
             assert.notStrictEqual(lastRequestUid, undefined);
             assert.strictEqual(lastRequestUid.indexOf(':'), -1);
             assert.strictEqual(ids.pop(), lastRequestUid);
         });
     });
 });
+
+describe('Sproxyd PUT error handling', function () {
+    this.timeout(10000);
+
+    const client = new Sproxy({
+        bootstrap: ['127.0.0.1:9001', '127.0.0.1:9000']
+    });
+    let serverWithError;
+    let server;
+
+    before('Create the server', done => {
```

Comment:
Please add another test testing the failover that it succeeds. Also a stress test ensuring requests do not fail will be nice

Reply:
the success failover  tests are performed as part of this block https://github.com/scality/sproxydclient/blob/d84a5a09b85b0dee55a78056b05c96fd29730679/tests/unit/sproxyd.js#L150